src/app/src/pages/redteam/setup/components/Targets/HttpEndpointConfiguration.tsx

```diff
@@ -112,6 +108,16 @@ Content-Type: application/json
   const [error, setError] = useState('');
   const [copied, setCopied] = useState(false);
 
+  // Auto-size the raw request textarea between 10rem and 40rem based on line count
+  const computeRawTextareaHeight = useCallback((text: string) => {
+    const lineCount = (text?.match(/\n/g)?.length || 0) + 1;
+    const lineHeightPx = 20; // approx for 14px monospace
+    const minPx = 10 * 16; // ~10rem
+    const maxPx = 40 * 16; // ~40rem
+    const desired = lineCount * lineHeightPx + 20; // padding allowance
+    return `${Math.min(maxPx, Math.max(minPx, desired))}px`;
+  }, []);
```

Comment:
this is a really weird approach- you should use css to do this IMO, not custom JS.